internal/pkg/vm/internal/vmcontext/vmcontext.go

```diff
@@ -135,7 +143,13 @@ func (ctx *VMContext) Send(to address.Address, method types.MethodID, value type
 		return nil, 1, errors.NewFaultErrorf("unhandled: sending to self (%s)", msg.From)
 	}
 
-	toActor, err := deps.GetOrCreateActor(context.TODO(), msg.To, func() (*actor.Actor, error) {
+	// fetch id address from init actor if necessary
+	toAddr, err := ctx.resolveActorAddress(msg.To)
```

Comment:
Why do we go through all of the trouble of doing this outside the state machine if we are just going to do it inside?  Why are both of these resolutions necessary?

Reply:
@ZenGround0 Because we are generating the VM context in two different places. I can't answer why we're doing that (history is gone). I rather we not do it in two places, but that's not a refactor I want to include in this PR.